packages/core/src/modules/credentials/protocol/v2/V2CredentialService.ts

```diff
@@ -220,8 +219,7 @@ export class V2CredentialService extends CredentialService {
     credentialRecord: CredentialExchangeRecord
   ): Promise<CredentialProtocolMsgReturnType<V2OfferCredentialMessage>> {
     const options: ServiceOfferCredentialOptions = {
-      connectionId: proposal.connectionId ?? undefined,
-      protocolVersion: proposal.protocolVersion,
+      protocolVersion: credentialRecord.protocolVersion,
```

Comment:
Why do we need protocolVersion here? It's already defined on the record and we're in v2 context here.